src/System.IO.FileSystem/tests/File/GetSetTimes.cs

```diff
@@ -108,6 +108,20 @@ public void SetLastWriteTimeTicks()
             Assert.True(firstFileTicks <= secondFileTicks, $"First File Ticks\t{firstFileTicks}\nSecond File Ticks\t{secondFileTicks}");
         }
 
+        [ConditionalFact(nameof(isNotHFS))] // OSX HFS driver format does not support nanosecond granularity
+        public void SetUptoNanoSeconds()
+        {
+            string firstFile = GetTestFilePath();
+            File.WriteAllText(firstFile, "");
+
+            DateTime dateTime = DateTime.UtcNow;
+            File.SetLastWriteTimeUtc(firstFile, dateTime);
+            long ticks = File.GetLastWriteTimeUtc(firstFile).Ticks;
+
+            Assert.Equal(dateTime, File.GetLastWriteTimeUtc(firstFile));
+            Assert.True(dateTime.Ticks == ticks, $"First File Ticks\t{ticks}\nSecond File Ticks\t{dateTime.Ticks}");
```

Comment:
Assert.Equal is not sufficient? Also that prints the two values for you if they don't match.